packages/datadog-plugin-aws-sdk/test/lambda.spec.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+const JSZip = require('jszip')
+const agent = require('../../dd-trace/test/plugins/agent')
+const plugin = require('../src')
+const { setup } = require('./spec_helpers')
+
+const zip = new JSZip()
+
+wrapIt()
+
+describe('Plugin', () => {
+  describe('aws-sdk (lambda)', function () {
+    setup()
+
+    withVersions(plugin, 'aws-sdk', version => {
+      let AWS
+      let lambda
+      let tracer
+
+      describe('without configuration', () => {
+        let ZipFile
+
+        before(async () => {
+          const lambdaFunctionCode = 'def handle(event, context):\n  return event\n'
+
+          zip.file('handler.py', lambdaFunctionCode.toString())
+          ZipFile = await zip.generateAsync({ type: 'nodebuffer' })
+        })
+
+        before(done => {
+          AWS = require(`../../../versions/aws-sdk@${version}`).get()
+
+          const lambdaEndpoint = new AWS.Endpoint('http://localhost:4566')
+          lambda = new AWS.Lambda({ endpoint: lambdaEndpoint, region: 'us-east-1' })
+
+          lambda.createFunction({
+            FunctionName: 'ironmaiden',
+            Code: { ZipFile },
+            Handler: 'handler.handle',
+            Role: 'arn:aws:iam::123456:role/test',
+            Runtime: 'python3.7'
+          }, (err, res) => {
+            if (err) return done(err)
+
+            agent.load('aws-sdk').then(done, done)
+          })
+          tracer = require('../../dd-trace')
+        })
+
+        after(done => {
+          lambda.deleteFunction({ FunctionName: 'ironmaiden' }, err => {
+            agent.close().then(() => done(err), done)
+          })
+        })
+
+        it('should propagate the tracing context with existing ClientContext', (done) => {
+          agent.use(traces => {
+            const span = traces[0][0]
+            const clientContextSent = Buffer.from(lambdaReq.params.ClientContext, 'base64').toString('utf-8')
+            const injectedTraceData = JSON.parse(clientContextSent).custom._datadog
+            const spanContext = tracer.extract('text_map', injectedTraceData)
+
+            expect(span.resource.startsWith('invoke')).to.equal(true)
+
+            const parentId = span.span_id.toString()
+            const traceId = span.trace_id.toString()
+            expect(spanContext.toTraceId()).to.equal(traceId)
+            expect(spanContext.toSpanId()).to.equal(parentId)
+          }).then(done, done)
+
+          const lambdaReq = lambda.invoke({
+            FunctionName: 'ironmaiden',
+            Payload: '{}',
+            ClientContext: 'eyJjdXN0b20iOnsieC1jb3JyZWxhdGlvbi10ZXN0LWN1aWQiOiJja2N4NGttNXUwMDAwMGNzM2NpbzdvODJsIn19'
+          }, e => e && done(e))
+        })
+
+        it('should propagate the tracing context without an existing ClientContext', (done) => {
```

Comment:
Should the other cases be tested as well?

Reply:
With the last test added I'm covering all logical paths: 
- Given ClientContext test
  - With `custom` key
  - With no `custom` key
 - Given no ClientContext test